### ISSUE TITLE
Allow disabling color codes in the output

### ISSUE THREAD

**Issue by tastyminerals** [o2sh/onefetch#1490]
### Summary 💡

It would be nice to have an option to disable shell color codes and make the output of **onefetch** easier to parse. 
For example
```
[31;1mCommits[0m[39;1m:[0m [39m411[0m
[31;1mChurn (17)[0m[39;1m:[0m [39m.travis.yml 6[0m
[39m            …/workflows/build-test.yml 4[0m
```
is problematic once you want to collect some stats across many repos.

Similar neofetch issue: https://github.com/dylanaraps/neofetch/issues/753

### Motivation 🔦

I want to collect repo summaries across many projects. 
```
ls -d * | xargs -I {} onefetch "{}" --no-bots --no-merges --number-of-authors 10 --number-of-file-churns 3 --no-art --no-color-palette --no-bold > summary.txt
```
produces a file which is difficult to parse without stripping the shell color codes first.

**Comment by o2sh** [comment]
Maybe you could use this option?

```
  -o, --output <FORMAT>
          Outputs Onefetch in a specific format

          [possible values: json, yaml]
```

**Comment by spenserblack** [comment]
Yeah, as @o2sh said, if you're looking for something machine-readable then you might want to use `-o json`. That, combined with `jq`, can help. E.g.
```shell
onefetch -o json | jq '.infoFields[].[]'
```

Though it might be reasonable to reinterpret the `--true-color auto|never|always` argument to `--color auto|true|ansi|never`.

**Comment by tastyminerals** [comment]
ok, `-o json` is a good workaround. 
the `--color never` or similar option could be a nice addition since one would expect the ability to disable colorization if `--color` exists.

**Comment by arimu1** [comment]
I'd like to take this on, referencing your gengo reference impl (spenserblack/gengo#491). Before I open a PR I want to check the design with you, since there are two viable approaches and they differ a lot in footprint:

**Surface:** replace `--true-color <auto|always|never>` with `--color <auto|always|ansi|never>` (default `auto`), keeping `--true-color` as a deprecated alias that warns on stderr — so existing scripts keep working. `always`/`ansi`/`auto` map onto the current true-color-vs-16-color pipeline; the new part is `never` fully disabling color.

**The `never` implementation is the open question:**
- **(A) Strip at the render choke point** — let the existing `owo_colors` styling run, then strip ANSI SGR escapes from the fully-rendered buffer at the single point all printer variants already funnel through, and skip the color-palette swatch. Small, localized diff; no churn across the ~8 independently-styled `Display` impls. Downside: it's a post-hoc strip rather than "never emit color."
- **(B) Thread a color-enabled flag through every call site** — plumb the setting into each `Display`/styling site so color is never generated. Truer to intent, but a large, invasive change touching title/languages/authors/ascii/palette/... .

I have (A) working (tests + manual verification that `--color never` emits zero escape bytes, `--color ansi` gives 16-color, `-o json` unaffected). Do you prefer (A), or would you rather I invest in (B)? Happy to go either way.

**Comment by arimu1** [comment]
Following up with two findings, since they narrow the A-vs-B question I asked above.

**The obvious shortcut doesn't exist.** I checked whether `owo_colors::set_override(false)` could just switch colour off globally — it can't. Per `owo-colors-4.3.0/src/overrides.rs`, the override only changes what `if_supports_color` does, and onefetch never calls that; every styled site goes through `.style(...)` / `.color(...)` directly (`info/title.rs`, `info/utils/mod.rs`, the per-field `Display` impls). So the override is a no-op here and there's no free version of (B).

**The gengo reference is (B), but it's cheap there for a structural reason.** In spenserblack/gengo#491 the whole thing lands in one function:

```rust
fn colorize(&self, s: &str, color: owo_colors::Rgb) -> String {
    match self.color {
        Never => String::from(s),
        Always => ...,
        Ansi => ...,
    }
}
```

gengo has a single colorize choke point, so "never emit colour" is a one-branch change. onefetch doesn't — colour is applied at ~8 independently-styled `Display` impls plus the ascii art and the palette swatch, which is exactly why (B) looked invasive when I scoped it.

That suggests a third option that's closer to your gengo design than either of mine:

**(B′)** introduce the choke point onefetch is missing — a small colorize/style helper that every field styles through, carrying the `--color` mode — then `Never` is one branch, same as gengo. Bigger than (A), much smaller than the full (B) plumbing, and it leaves onefetch with the structure your reference impl assumes.

I have (A) working on a branch already (`--color auto|always|ansi|never`, `--true-color` kept as a deprecated alias, verified zero escape bytes under `never` and 16-colour under `ansi`). Happy to convert it to (B′) instead — just say which and I'll open the PR. If you'd rather leave this closed given `-o json` covers the original use case, that's a fine answer too and I'll drop it.
